tests/optim/test_oss.py

```diff
@@ -263,6 +262,25 @@ def test_zero_grad():
     mp.spawn(run_test_zero_grad, args=(world_size, temp_file_name), nprocs=world_size, join=True)
 
 
+def run_test_catch_empty_shardd(rank, world_size, tempfile_name):
+    dist_init(rank, world_size, tempfile_name, backend="gloo")
+    x = torch.rand(1)
+    m = torch.nn.Linear(1, 1)
+    try:
```

Comment:
Can we use assertRaises if you are trying to confirm that an error is raised?

Reply:
sure, all test frameworks have their own checks but can do. I think that one promise of pytest is to limit these actually